### PR TITLE
lint/remark frail

### PR DESCRIPTION
**Pull request by maxwondercorn**
Added `--frail` option to `npm run lint:md` so linting issues will throw errors in CI.  

- Bumped ember-dictionary version
- Linted and edited README.md and CONTRIBUTING.md for consistency with the rest of the guides

This PR based on discussion in #79.

### CODE REVIEW

**Comment by maxwondercorn** [comment]
Sorry for the confusing PR.  Reused the dead links linting branch, cleaned it up and created a new PR from it.

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "ember-cli-template-lint": "^1.0.0-beta.1",
     "ember-cli-uglify": "^2.1.0",
     "ember-data": "~3.5.0",
-    "ember-dictionary": "0.1.0",
+    "ember-dictionary": "0.1.3",
```
Comment (Gaurav0) [inline package.json]: Should we add a ~ or ^ here?

Comment (maxwondercorn) [reply package.json]: missed that.  ^ will work as it just's list of words.  No breaking code... I will update pr.

**Comment by jenweber** [comment]
Thank you! I ran this locally and it looks great. Merged.

**Comment by maxwondercorn** [comment]
Would you like me to implement this on the other guides?

**Comment by jenweber** [comment]
That would be awesome! Is there a way for us to run these tests on the diff of new PRs only? The Guides are massive and will have a thousand things to spellcheck or add as exceptions.

**Comment by maxwondercorn** [comment]
Not sure, I'd have to look into it to see what's possible.  The worst case for spell checking would be the initial setup.  It would catch the technical words that aren't already in the dictionary.  I'd think I got most of the potential words in the dictionary based on the cli-guides.  After that it would only be checking PRs.

For the ember guides proper, you'd only spell check the current release and upcoming release guides.  All earlier versions would be put into the ignore file

**Pull request merged**